### PR TITLE
[MM-48031] Work template read API

### PR DESCRIPTION
**Pull request by JulienTant**
#### Summary

- Implements the work template generator, creating a go file from the yaml template.
  - Like for the translation files (and all of our other generated files), the content is validated by the CI to ensure that the new `make` command has been run whenever the work templates yaml changes.
  - Generated templates are not meant to be used directly, this is why they are private.
- Implements the read REST API for retrieving categories and associated templates.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-48031

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: app/worktemplates/templates.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      id: "board-project-task"
+      template: "project task|bmttiziw35irgtmztewd9upyqdy"
+      name: project task board
+      channel: feature-release
+      illustration: "https://via.placeholder.com/509x352.png?text=Board+project+task"
+  - playbook:
+      template: "product release" # <-- playbooks templates don't have ids, have to rely on name
+      name: "Feature release"
+      id: product-release-playbook
+      illustration: "https://via.placeholder.com/509x352.png?text=Playbook+feature+release"
```
Comment (julmondragon) [inline app/worktemplates/templates.yaml:42]: What's the plan for the definitive location of these illustrations?

**Review by julmondragon — APPROVED** [review]
LGTM @JulienTant ! Solid work on this, good stuff.

File: app/worktemplates/templates.yaml (hunk 1 above)
Comment (JulienTant) [reply app/worktemplates/templates.yaml:42]: They're going to be hosted on mattermost-webapp, so those URLs will be relative to the root of the workspace `/images/worktemplates/{path_to_final_pic}.png`

**Review by mhd-sln — APPROVED** [review]
LGTM! Thanks @JulienTant!

**Pull request merged**